src/main/java/org/truffleruby/parser/lexer/RubyLexer.java

```diff
@@ -2960,7 +2961,7 @@ public boolean tokadd_mbchar(int first_byte) {
 
         if (length <= 0) {
             compile_error("invalid multibyte char (" + getEncoding() + ")");
-        } else if (length > 1) {
+        } else if (length > 1 || (tokenCR == CR_7BIT && first_byte > 127)) {
```

Comment:
It might be nice to have a utility function for this. E.g. `!isSingleByteChar(first_byte)`

Reply:
The bug here was the character was single byte because the encoding was `ASCII-8BIT` (all chars are single byte with that encoding). But I can add something similar.

Reply:
If this utility is added, there's other places that could use it which are currently `< 128` conditions.